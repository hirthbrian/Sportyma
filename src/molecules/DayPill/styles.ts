import styled from "styled-components/native";

import { KanitText } from "../../atoms";

export const Container = styled.View`
  background-color: #2559f2;
  border-radius: 30px;
`;

export const Day = styled(KanitText)`
  color: #ffffff;
  letter-spacing: 1px;
  padding: 4px 10px;
`;
