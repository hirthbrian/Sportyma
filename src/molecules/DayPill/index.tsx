import { format } from "date-fns";

import { Container, Day } from "./styles";

import { Props } from "./types";

const DayPill = ({ date }: Props) => {
  return (
    <Container>
      <Day style={{ fontSize: 12 }}>
        {format(new Date(date), "d MMMM").toUpperCase()}
      </Day>
    </Container>
  );
};

export default DayPill;
