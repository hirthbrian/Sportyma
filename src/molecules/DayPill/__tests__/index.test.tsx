import React from "react";
import renderer from "react-test-renderer";

import DayPill from "..";

const date = "2023-02-16T20:35:00+00:00" as unknown as Date;

describe("<DayPill />", () => {
  let props;

  beforeEach(() => {
    props = {
      date,
    };
  });

  it("renders correcly", () => {
    expect(renderer.create(<DayPill {...props} />).toJSON()).toBeTruthy();
  });
  it("match snapshot", () => {
    const tree = renderer.create(<DayPill {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
