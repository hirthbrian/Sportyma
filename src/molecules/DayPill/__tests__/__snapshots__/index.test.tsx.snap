// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DayPill /> match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#2559f2",
        "borderRadius": 30,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "Kanit-Regular",
          "fontSize": 16,
          "letterSpacing": 1,
          "paddingBottom": 4,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 4,
        },
        Object {
          "fontSize": 12,
        },
      ]
    }
  >
    16 FEBRUARY
  </Text>
</View>
`;
